From ae6824cd9601cc41796b277d365f931a56e7721a Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 6 Jan 2014 21:02:35 -0500
Subject: [PATCH 0140/1936] Fix OSX Portability Issues

- Confirmed fix for missing pthread headers
- Unconfirmed fix for eliminating linux/bsd style threading priority code
---
 src/CMakeLists.txt  | 9 +++++++--
 src/Misc/Master.h   | 2 ++
 src/Misc/Part.h     | 1 +
 src/Nio/SafeQueue.h | 1 +
 4 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ddc0fd7f..786231a7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -62,8 +62,13 @@ SET (LibloEnable ${LIBLO_FOUND} CACHE BOOL
 # Add version information
 add_definitions(-DVERSION="${VERSION}")
 
-# Add scheduler function existance info (OSX compatiability)
-add_definitions(-DHAVE_SCHEDULER=${HAVE_SCHEDULER})
+message(STATUS "Building on a '${CMAKE_SYSTEM_NAME}' System")
+
+if(NOT "Darwin" STREQUAL ${CMAKE_SYSTEM_NAME})
+    # Add scheduler function existance info (OSX compatiability)
+    add_definitions(-DHAVE_SCHEDULER=${HAVE_SCHEDULER})
+endif()
+
 
 # Give a good guess on the best Input/Output default backends
 if (JackEnable)
diff --git a/src/Misc/Master.h b/src/Misc/Master.h
index 0de022b5..b0179097 100644
--- a/src/Misc/Master.h
+++ b/src/Misc/Master.h
@@ -23,6 +23,8 @@
 
 #ifndef MASTER_H
 #define MASTER_H
+#include <pthread.h>
+
 #include "../globals.h"
 #include "Microtonal.h"
 
diff --git a/src/Misc/Part.h b/src/Misc/Part.h
index 4221f289..ed247d30 100644
--- a/src/Misc/Part.h
+++ b/src/Misc/Part.h
@@ -29,6 +29,7 @@
 #include "../Params/Controller.h"
 #include "../Misc/Microtonal.h"
 
+#include <pthread.h>
 #include <list> // For the monomemnotes list.
 
 class EffectMgr;
diff --git a/src/Nio/SafeQueue.h b/src/Nio/SafeQueue.h
index 7b41166b..072a1e09 100644
--- a/src/Nio/SafeQueue.h
+++ b/src/Nio/SafeQueue.h
@@ -3,6 +3,7 @@
 #define SAFEQUEUE_H
 #include <cstdlib>
 #include <semaphore.h>
+#include <pthread.h>
 
 /**
  * C++ thread safe lockless queue
-- 
2.47.0

